### PR TITLE
Fix saving of cookies

### PR DESCRIPTION
**Pull request by TheOneRing**
Fixes: #7700

And again we need to discuss the change log

### CODE REVIEW

**Comment by TheOneRing** [comment]
@micbar and how am I supposed to reference the pr before I create a review?
Do I need to create a issue if there was none?
Can I just use a random name instead?

**Comment by micbar** [comment]
Yes. This has some limitations.

Calens needs one primary issue id. It is always the id of the first url in the list.

That could be an issue that you create before creating the pr.
This issue could point then to the pr.
The url should be github.com/vendor/repo/issues/1234  which BTW always resolves both types, issues and PRs.

**Comment by TheOneRing** [comment]
Yes we now use the changelog.md but it's only generated on Windows

**Comment by micbar** [comment]
> Yes we now use the changelog.md but it's only generated on Windows

Windows?

**Comment by dschmidt** [comment]
Are we losing the old changelog completely? Can we somehow migrate the old changelog to the new autogenerated one?

**Comment by micbar** [comment]
> Are we losing the old changelog completely? Can we somehow migrate the old changelog to the new autogenerated one?

Already done. See on master.

File: test/testcookies.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+        QNetworkCookie cookieA = QNetworkCookie("foo", "bar");
+        // tomorrow rounded
+        cookieA.setExpirationDate(QDateTime(QDateTime::currentDateTimeUtc().addDays(1).date()));
+        const QList<QNetworkCookie> cookies = {cookieA, QNetworkCookie("foo2", "bar")};
+        CookieJar jar;
+        jar.setAllCookies(cookies);
+        QCOMPARE(cookies, jar.allCookies());
+        QVERIFY(jar.save(tmp.filePath("test.db")));
+        QVERIFY(jar.save(nonexistingPath));
+
```
Comment (dschmidt) [inline test/testcookies.cpp]: At first it was a bit confusing to me why you saved to two different paths and only verified loading only in one of them afterwards... maybe leave a comment what you are testing exactly.

**Comment by guruz** [comment]
Please note that the old hand-made changelog contains (well, should) REAL CHANGES in USER UNDERSTANDABLE language and not just random git log messages.

**Comment by TheOneRing** [comment]
> Please note that the old hand-made changelog contains (well, should) REAL CHANGES in USER UNDERSTANDABLE language and not just random git log messages.

As you can see in https://github.com/owncloud/client/blob/master/CHANGELOG.md the new change log does it too, it is supposed to be even more detailed.

**Comment by micbar** [comment]
@guruz What brought you to the conclusion that we are using git log messages?

**Comment by TheOneRing** [comment]
If the change log is sufficient this one is ready to be merged?

**Comment by TheOneRing** [comment]
ping

**Pull request merged**